Add plugin: Featured Image

johansan opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/johansan/obsidian-featured-image

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[const youtubeFolder = `${thumbnailFolder}/youtube`;](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/main.ts#L354-L355)
[Use `normalizePath` to clean up user defined paths](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+%60normalizePath()%60+to+clean+up+user-defined+paths)

['Update All Featured Images',](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/main.ts#L484), ['Remove All Featured Images',](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/main.ts#L521)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[export class WelcomeModal extends Modal {](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/modals.ts#L53-L101)
tbh, that modal makes no sense, it's the same content as in the README, which uses have just read prior to installing the plugin.
Remove it.

[containerEl.createEl("h1", { text: "Featured Image" });](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/settings.ts#L63-L81)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
And the buy me a coffee link should be at the bottom, not at the start too.

[src: "https://cdn.buymeacoffee.com/buttons/v2/default-yellow.png",](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/settings.ts#L78)
This needs to be embedded into the plugin, not loaded externally.

[.setName('General settings')](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/settings.ts#L84)
Same as above.

[.setName('Frontmatter settings')](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/settings.ts#L112), [.setName('Youtube settings')](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/settings.ts#L140), [.setName('Local media settings')](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/settings.ts#L167), [.setName('Developer options')](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/settings.ts#L207)
[Avoid "settings" and "options" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

[.setName('Youtube settings')](https://github.com/johansan/obsidian-featured-image/blob/62fd10460ad94d4e36e62668f778378ee237aa22/src/settings.ts#L140)
Use the correct capitalization of YouTube.

johansan commented:
@joethei Many thanks for your very detailed review and instructions on how to correct the issues you found.

Much appreciated!!

I have published Featured Image 1.0.3 which solves all these issues:

1. normalizePath
processAutoCardLinkImage() and downloadThumbnail() now use normalizePath() to clean up user-defined paths.

2. Settings dialog
Now uses "Sentence case" in all places

4. Modal dialog box "Welcome"
Removed, I agree with your comments 100%.

6. "Buy me a coffee" section in Settings
Removed completely including the external image. 

8. Label "General settings" in top of Settings
Removed this "top section" including descriptive text. 

10. Sections in Settings such as "Frontmatter settings", "Developer options"
Renamed to "Frontmatter", "Developer", etc.

12. Capitalization of "Youtube"
Renamed to "YouTube" in all places.

Thanks for your time!

Johan
